migrate-3-to-4/exporter.sh

```diff
@@ -0,0 +1,211 @@
+        echo "Fetching pod logs -"
+        kubectl  -n "$namespace" logs "$(kubectl -n "$namespace" get pods -l app=circle-migrator -o name)" > "$path"/logs/circle-migrator.log
+        echo "Pod log is available at $path/logs/circle-migrator.log"
+        echo "Removing job/circle-migrator -"
+        kubectl delete job/circle-migrator --namespace "$namespace"
+    else
+        echo "Status wait timeout for job/circle-migrator, Check the Log via - kubectl logs pods -l app=circle-migrator"
+        echo "Job circle-migrator will delete automatically after 24 hours once complete."
+    fi
+}
+
+export_postgres() {
+    echo "Exporting PostgreSQL data"
+    PG_POD=$(kubectl -n "$namespace" get pods | grep postgresql | tail -1 | awk '{print $1}')
+    PG_PASSWORD=$(kubectl -n "$namespace" get secrets postgresql -o jsonpath="{.data.postgresql-password}" | base64 --decode)
+    kubectl -n "$namespace" exec -it "$PG_POD" -- bash -c "export PGPASSWORD='$PG_PASSWORD' && pg_dumpall -U postgres -c" > ${BACKUP_DIR}/circle.sql
+}
+
+##
+# Basic sanity check / smoke test to ensure that the postgresql export performed by
+# the export_postgres function contains what we expect it to contain
+##
+check_postgres() {
+    echo "Verifying postgres export file"
+    if [ -z "$(grep build_jobs ${BACKUP_DIR}/circle.sql | head -n1)" ]
+    then
+        echo "[FATAL] Something is wrong with the postgresql export file for 'conductor_production' database, please contact CircleCI support at enterprise-support@circleci.com for further assistance."
+        exit 1
+    fi
+    if [ -z "$(grep contexts ${BACKUP_DIR}/circle.sql | head -n1)" ]
+    then
+        echo "[FATAL] Something is wrong with the postgresql export file for 'contexts_service_production' database, please contact CircleCI support at enterprise-support@circleci.com for further assistance."
+        exit 1
+    fi
+    if [ -z "$(grep qrtz_blob_triggers ${BACKUP_DIR}/circle.sql | head -n1)" ]
+    then
+        echo "[FATAL] Something is wrong with the postgresql export file for 'cron_service_production' database, please contact CircleCI support at enterprise-support@circleci.com for further assistance."
+        exit 1
+    fi
+    if [ -z "$(grep tasks ${BACKUP_DIR}/circle.sql | head -n1)" ]
+    then
+        echo "[WARN] 'vms' database was not correctly exported."
+    fi
+}
+
+export_mongo() {
+    echo "Exporting MongoDB data"
+    MONGO_POD="mongodb-0"
+    MONGODB_USERNAME="root"
+    MONGODB_PASSWORD=$(kubectl -n "$namespace" get secrets mongodb -o jsonpath="{.data.mongodb-root-password}" | base64 --decode)
+    TEMP_DIR="/bitnami/circle-mongo"
+    kubectl -n "$namespace" exec -it "$MONGO_POD" -- bash -c "mkdir $TEMP_DIR"
+    kubectl -n "$namespace" exec -it "$MONGO_POD" -- bash -c "mongodump -u '$MONGODB_USERNAME' -p '$MONGODB_PASSWORD' --authenticationDatabase admin --db=circle_ghe --out=$TEMP_DIR"
+    kubectl -n "$namespace" cp $MONGO_POD:$TEMP_DIR ${BACKUP_DIR}/circle-mongo
```

Comment:
could we include a step here to remove the /bitnami/circle-mongo/ backup directory
just in case the script needs to be rerun and the local mongobackup was deleted. not a blocker, just a thought